### ISSUE TITLE
Ghost <-> Ghost Admin Repo Split Tooling

### ISSUE THREAD

**Issue by ErisDS** [TryGhost/Ghost#6977]
This issue is intended to be used for tracking all the tasks that remain in terms of needing tooling post repo split. @kevinansfield please feel free to edit this message as needed!

There are some small and some big tasks remaining as part of the repo split. They fall roughly into two categories - build tooling (largely being done by @kevinansfield at the moment but I'm sure he'd love some help) and release tooling, which is largely about making sure we can credit contributors to Ghost-Admin. We'd really really ❤️  some help to do this.

#### Build About Page

- [ ] Move buildAboutPage & update it to count contributors from both repos

The `buildAboutPage` task currently lives in Ghost's grunt file, but as it creates templates and files for the client, it should really live in the Ghost-Admin grunt file. This task needs updating to pull down top contributors from both repositories, and combine the data. It's probably going to get much bigger, and might need moving out to its own repository.

#### Contributor list

- [ ] Create a tool (grunt task?) to output the list of contributors for the release post.

Right now, we use this git alias to output a nice comma separated list of people who contributed, ordered by commits (not including merges):
`contribs = "!f() { git shortlog -sn --no-merges $1.. | awk '{printf (\"%s %s, \", $2, $3)}'; }; f"` 

This approach isn't going to work now that Ghost-Admin is split out, so we need some other way to grab the names to put them in the release process. 

##  Build Tools & Cleanup:

- [ ] Add bower to Ghost-Admin
- [ ] Add a new task (`grunt init` or `grunt build`?) to Ghost-Admin grunt file for installing dependencies
- [ ] Remove bower from Ghost (PR open - #6910)
- [ ] Remove old ember-related JSCS rules from Ghost (PR open - #6910)
- [ ] Add a new task (`grunt build` ?) to Ghost which does NOT do submodule deps but does build everything. 

#### New grunt task for managing both repos

To explain the idea behind the new `grunt build` (or whatever it gets named) task, it should be for developers (where `grunt init` will be only for installing now) to setup their environment for developing. Ideally it would take two PR numbers, to checkout & build specific PRs for both the Ghost and Ghost admin repos. 

E.g. for scheduling, I might want to run `grunt build --g=6861 --ga=25`. This would set Ghost to PR 6861 and Ghost-Admin to PR 25. It would then run all the build tasks so that your repository was setup ready to test out the scheduler feature. Not that for this to work, it would probably require an update to your git config for both repos, adding `fetch = +refs/pull/*/head:refs/remotes/upstream/pr/*` to upstream as [described here](https://dev.ghost.org/easy-git-pr-test/).

**Comment by acburdine** [comment]
Just as a thought (since we're doing contribs for ghost and ghost-admin now), do we want to add contribs for Casper as well or no?

**Comment by acburdine** [comment]
Another thought (something that should be added to the buildAboutPage task): Because we do a lot of PR's that are from GreenKeeper, it shows up as a top contributor. That's somewhat misleading, so we might want to add a clause to prevent that particular committer from showing up.

**Comment by acburdine** [comment]
@ErisDS @kevinansfield I discovered something whilst working on this that will make the numbers skewed a bit.

There are duplicate commits between the two Ghost-Admin and Ghost repositories (the ones pertaining to the client), and because the commit shas are different, there's no easy way to compare the two commits to see if they are the same short of comparing the commit messages.

This will only occur until 90 days after the split, but I guess my question is should I have the commit messages be compared (would take edits to top-gh-contribs) or just go with the skewed number counts (for this release only most likely)?

**Comment by ErisDS** [comment]
> This will only occur until 90 days after the split, but I guess my question is should I have the commit messages be compared (would take edits to top-gh-contribs) or just go with the skewed number counts (for this release only most likely)?

I'm happy to live with the duplication as its ownly short term, I think it's a fiddly thing to fix?

**Comment by acburdine** [comment]
@ErisDS yeah trying to fix it is probably more trouble than it's worth :smile:
